backvendor/filehash.go

```
@@ -29,38 +27,36 @@ import (
 // version control system that tracks it.
 type FileHash string
 
-// FileHashes is a map of paths, relative to the top-level of the
-// version control system, to their hashes.
-type FileHashes struct {
-	vcsCmd, root string
-	hashes       map[string]FileHash
+// Hasher is the interface that wraps the Hash method.
+type Hasher interface {
+	// Hash returns the file hash for the filename absPath, hashed
+	// as though it were in the repository as filename
+	// relativePath.
+	Hash(relativePath, absPath string) (FileHash, error)
 }
 
-func hashFile(vcsCmd, relativePath, absPath string) (FileHash, error) {
-	if vcsCmd != vcsGit {
-		return FileHash(""), ErrorUnknownVCS
-	}
-
-	args := []string{"hash-object", "--path", relativePath, absPath}
-	cmd := exec.Command(vcsCmd, args...)
-	var buf bytes.Buffer
-	cmd.Stdout = &buf
-	cmd.Stderr = &buf
-	err := cmd.Run()
-	if err != nil {
-		os.Stderr.Write(buf.Bytes())
-		return FileHash(""), err
+func NewHasher(vcsCmd string) Hasher {
+	switch vcsCmd {
+	case vcsGit:
+		return &gitHasher{}
 	}
+	return nil
```

Why not return an error or panic here?

---

I'll have NewHasher return a bool for 'ok', similar to a map look-up.

---

Updated.